source/src/plugins/Chi2Test.cc

```diff
@@ -0,0 +1,147 @@
+#include <dqm4hep/Logging.h>
+#include <dqm4hep/MonitorElement.h>
+#include <dqm4hep/PluginManager.h>
+#include <dqm4hep/QualityTest.h>
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/XmlHelper.h>
+#include <dqm4hep/AnalysisHelper.h>
+
+// -- root headers
+#include <TH1.h>
+#include <TMath.h>
+
+namespace dqm4hep {
+
+  namespace core {
+
+    /** Chi2Test class
+     */
+    class Chi2Test : public QualityTest {
+    public:
+      class Factory : public QualityTestFactory {
+      public:
+        QTestPtr createQualityTest(const std::string &qname) const override;
+      };
+
+      Chi2Test(const std::string &name);
+      ~Chi2Test() override = default;
+      StatusCode readSettings(const dqm4hep::core::TiXmlHandle xmlHandle) override;
+      std::string getTestOptions(std::string comparisonType, bool useUnderflow, bool useOverflow);
+      void userRun(MonitorElement* monitorElement, QualityTestReport &report) override;
+
+    protected:
+      std::string m_comparisonType;
+      bool m_useUnderflow = {false};
+      bool m_useOverflow = {false};
+
+    };
+
+    typedef Chi2Test::Factory Chi2TestFactory;
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    inline QTestPtr Chi2Test::Factory::createQualityTest(const std::string &qname) const {
+      return std::shared_ptr<QTest>(new Chi2Test(qname));
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    Chi2Test::Chi2Test(const std::string &qname)
+        : QualityTest("Chi2", qname),
+	  m_comparisonType("UU"),
+	  m_useUnderflow(),
+	  m_useOverflow()
+{
+      m_description = "Performs the chi-squared test on a TH1 and a reference, outputting the p-value. This is analogous to the Kolmogorov-Smirnov "
+	              "test but is appropriate for use on binned data like histograms, which the Kolmogorov test is not.";
+    }
+
+    //-------------------------------------------------------------------------------------------------
+
+    StatusCode Chi2Test::readSettings(const TiXmlHandle xmlHandle) {
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "ComparisonType", m_comparisonType));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "UseUnderflow", m_useUnderflow));
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "UseOverflow", m_useOverflow));
+
+      return STATUS_CODE_SUCCESS;
+    }
+
+    std::string Chi2Test::getTestOptions(std::string comparisonType, bool useUnderflow, bool useOverflow) {
+
+      std::string optionsString = comparisonType;
```

Comment:
You don't protect aginst anything here. I can still pass `<parameter name="ComparisonType" value="CHI2" />` it will be accepted. Check what the string is what you expect